core/src/calleeFunctions/helpers/uniswapV3.ts

```diff
@@ -76,7 +76,7 @@ export const convertCollateralToDaiUsingPool = async function (
     if (!route) {
         throw new Error(`No route found for ${collateralSymbol} to DAI`);
     }
-    const quote =  UNISWAP_FEE
+    const quote = quoteGasAdjusted ? Math.ceil(quoteGasAdjusted.toNumber()) : UNISWAP_FEE;
```

Comment:
so if i change this to hardcoded `UNISWAP_FEE`, then the ui displays the value. if not, then autorouting extraction fails and in the ui the `Unknown` is displayed.

The error is not very informative:
```
Error: call revert exception [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="quoteExactInput(bytes,uint256)", data="0x", errorArgs=null, errorName=null, errorSignature=null, reason=null, code=CALL_EXCEPTION, version=abi/5.6.4)
```

this one appears in multitude of the cases. Including (as of my xp) the cases where the provided encoded route includes a duplicate at the end (e.g. `[... 'DAI', 'DAI]`)

Reply:
Those are two completely different things: fee for the transaction paid as gas and fee for the exchange paid to the protocol. The second thing is sometimes called `pool` for the Uniswap v3 since there are different liquidation pools that are identified by the fee that pool takes. TLDR: you need to find another kind fee inside `getRouteAndGasQuote`